### PR TITLE
[OpenBLAS_jll] Upgrade to new build optimised for PowerPC

### PR DESCRIPTION
**Pull request by giordano**
This only difference compared to previous build is that this one enables use of dynamic architecture detection also for the PowerPC architecture.

Ref: https://discourse.julialang.org/t/building-openblas-for-powerpc-defaults-to-power8/93972.

### CODE REVIEW

**Comment by vchuravy** [comment]
Can you also update the source recipe here?

**Comment by giordano** [comment]
Uhm, where exactly?  In https://github.com/JuliaLang/julia/blob/80dc13595c0aa29062705547debab679b5735c08/deps/openblas.mk I don't see anything PowerPC-specific (apart from a patch).  Maybe I should just delete https://github.com/JuliaLang/julia/blob/e536c77f4dc693aafc48af910b4fd86b487e900d/Make.inc#L834-L835 to allow optimising for the current architecture as we do on Intel?

**Comment by ViralBShah** [comment]
That seems about right.

**Comment by ViralBShah** [comment]
Not relevant for this PR, but it would be nice for all the logic to set the various OPENBLAS build options to live in openblas.mk.

**Comment by ViralBShah** [comment]
If you just don't set `OPENBLAS_TARGET_ARCH` explicitly, we do default to `DYNAMIC_ARCH=1` (higher up in Make.inc). May be best to just delete those two lines.

**Comment by giordano** [comment]
Ok, done!  To be honest I was a bit lost among those options :sweat_smile:

**Comment by ViralBShah** [comment]
Yeah, it is quite a maze. it would be nice to refactor those and put them all in openblas.mk.

**Comment by vchuravy** [comment]
Running `make -C deps compile-openblas` I see tuning for power9 and https://github.com/xianyi/OpenBLAS/blob/10be02c89605334b3095c0ac76f4aa328ae54f72/cmake/arch.cmake#L57-L58 does confirm that this is correct.

Nice!

**Review by vchuravy — DISMISSED** [review]

**Comment by vchuravy** [comment]
I spoke to soon.

It looks like we only built Power9...

```
/bin/sh: line 1: 609325 Segmentation fault      (core dumped) OPENBLAS_NUM_THREADS=1 OMP_NUM_THREADS=1 ./test_sbgemm > SBBLAT3.SUMM
```

File: Make.inc [hunk 1]
```diff
@@ -831,8 +831,6 @@ endif
 # If we are running on powerpc64le or ppc64le, set certain options automatically
 ifneq (,$(filter $(ARCH), powerpc64le ppc64le))
 JCFLAGS += -fsigned-char
-OPENBLAS_DYNAMIC_ARCH:=0
-OPENBLAS_TARGET_ARCH:=POWER8
```
Comment (vchuravy) [inline Make.inc:835]: We want this to still be POWER8. From the openblas docs

> The TARGET option can be used in conjunction with DYNAMIC_ARCH=1 to specify which cpu model should be assumed for all the common code in the library, usually you will want to set this to the oldest model you expect to encounter. Please note that it is not possible to combine support for different architectures, so no combined 32 and 64 bit or x86_64 and arm64 in the same library.

**Comment by vchuravy** [comment]
Hm still getting a seqfault on Power8, but this time I actually saw it build many Power6, Power8, and Power9...

```
/bin/sh: line 1: 714674 Segmentation fault      (core dumped) OPENBLAS_NUM_THREADS=1 OMP_NUM_THREADS=1 ./test_sbgemm > SBBLAT3.SUMM
make[2]: *** [Makefile:167: level3] Error 139
make[2]: Leaving directory '/nobackup/users/vchuravy/dev/julia/deps/scratch/openblas-b89fb708caa5a5a32de8f4306c4ff132e0228e9a/test'
make[1]: *** [Makefile:149: tests] Error 2
make[1]: Leaving directory '/nobackup/users/vchuravy/dev/julia/deps/scratch/openblas-b89fb708caa5a5a32de8f4306c4ff132e0228e9a'
```

**Comment by RajalakshmiSR** [comment]
What is the compiler version and make command used to build openblas? Are you trying it on POWER8 or POWER9?

**Comment by vchuravy** [comment]
This was with:

```
make USE_BINARYBUILDER_OPENBLAS=0 -C deps compile-openblas
```

```
[vchuravy@satori-login-001 ~]$ gcc --version
gcc (GCC) 8.3.1 20191121 (Red Hat 8.3.1-5)
Copyright (C) 2018 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```

```
[vchuravy@satori-login-001 ~]$ cat /proc/cpuinfo 
processor	: 0
cpu		: POWER9, altivec supported
clock		: 3800.000000MHz
revision	: 2.2 (pvr 004e 1202)
```

Slightly surprised thought this was a Power8 cluster xD

**Comment by RajalakshmiSR** [comment]
> This was with:
> 
> ```
> make USE_BINARYBUILDER_OPENBLAS=0 -C deps compile-openblas
> ```
> 
> ```
> [vchuravy@satori-login-001 ~]$ gcc --version
> gcc (GCC) 8.3.1 20191121 (Red Hat 8.3.1-5)
> Copyright (C) 2018 Free Software Foundation, Inc.
> This is free software; see the source for copying conditions.  There is NO
> warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
> ```
> 
> ```
> [vchuravy@satori-login-001 ~]$ cat /proc/cpuinfo 
> processor	: 0
> cpu		: POWER9, altivec supported
> clock		: 3800.000000MHz
> revision	: 2.2 (pvr 004e 1202)
> ```
> 
> Slightly surprised thought this was a Power8 cluster xD

Thanks, I can recreate it. Will debug this.

**Comment by RajalakshmiSR** [comment]
This is fixed here https://github.com/xianyi/OpenBLAS/pull/3718

**Comment by RajalakshmiSR** [comment]
openblas 0.3.22 will have the fix included.

**Comment by KristofferC** [comment]
Good to go now?

**Comment by RajalakshmiSR** [comment]
Thanks. looks good.

**Pull request merged**